src/Microsoft.AspNetCore.Routing/Builder/EndpointRoutingApplicationBuilderExtensions.cs

```diff
@@ -8,27 +8,33 @@
 
 namespace Microsoft.AspNetCore.Builder
 {
-    public static class GlobalRoutingApplicationBuilderExtensions
+    public static class EndpointRoutingApplicationBuilderExtensions
     {
-        private const string GlobalRoutingRegisteredKey = "__GlobalRoutingMiddlewareRegistered";
+        private const string EndpointRoutingRegisteredKey = "__EndpointRoutingMiddlewareRegistered";
 
+        [Obsolete("Replaced with UseEndpointRouting.")]
```

Comment:
Will other projects build when an `ObsoleteAttribute` is encountered?

Reply:
No, we have warnings as errors.